src/fastmcp/server/server.py

```
@@ -833,151 +845,171 @@ async def _get_resource_or_template_or_none(
-            except NotFoundError:
-                continue
+        results = await gather(
+            *[p.get_resource(uri) for p in self._providers],
+            return_exceptions=True,
+        )
+
+        for result in results:
+            if isinstance(result, Resource) and self._is_component_enabled(result):
+                return result
 
         raise NotFoundError(f"Unknown resource: {uri}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)